substrate/frame/contracts/src/lib.rs

```diff
@@ -230,7 +230,7 @@ pub mod pallet {
 	pub struct Pallet<T>(_);
 
 	#[pallet::config]
-	pub trait Config: frame_system::Config {
+	pub trait Config: frame_system::Config + pallet_xcm::Config {
```

Comment:
This requirement means that in the future, the contracts pallet cannot function without the XCM pallet being present. Could we somehow loosely couple the XCM pallet instead?

Reply:
yes that's an option I considered as well, if we can extract all the trait functions from a primitive crate that would be worth it, else I can move it behind a feature flag.
Keeping it like that for now to iterate quickly but will revisit before marking it as ready for final reviews